docs: Move render target to main Makefile as 'render-docs'

jrajahalme opened the pull request:
Building docs-container failed if built after building cilium docker
image ('make docker-image') due to the .dockerignore file produced for
building the cilium image being accidentally reused for building
docs-container. Fix this by explicitly creating a .dockerignore file
as part of the docs-container actions.

Move the 'render' target from Documentation folder to the main
Makefile as 'render-docs' and make it depend on the 'docs-container'
target.

Reorganize Documentation/Dockerfile so that any repo changes outside
of Documentation/requirements.txt will use the cached build toolchain,
making repeated 'make render-docs' commands run faster.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aalemayhu reviewed (changes requested):
LGTM, but the 'Building Documentation' section needs to be updated. Could you do it in this pull request? Thanks.

jrajahalme committed:
docs: Move render target to main Makefile as 'render-docs'

Building docs-container failed if built after building cilium docker
image ('make docker-image') due to the .dockerignore file produced for
building the cilium image being accidentally reused for building
docs-container. Fix this by explicitly creating a .dockerignore file
as part of the docs-container actions.

Move the 'render' target from Documentation folder to the main
Makefile as 'render-docs' and make it depend on the 'docs-container'
target.

Reorganize Documentation/Dockerfile so that any repo changes outside
of Documentation/requirements.txt will use the cached build toolchain,
making repeated 'make render-docs' commands run faster.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
Documentation: Update building documentation section.

Update the "Building Documentation" section to match the new Makefiles.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
